localgovdrupal/localgov_base
# Converts CSS to logical properties

Closes #628 

## What does this change?

Removes all instances of things like `margin-left`, `margin-right`, etc and use logical properties for them instead, such as `margin-inline-start` and `margin-inline-end`. 

This will be very beneficial for any site that wants to use languages other than left-to-right languages. It will be even more beneficial if we scale the project beyond UK/Ireland. Oh, and it will also mean a smaller CSS codebase to maintain.

## How to test

- Checkout this branch
- Load the demo content module
- Make sure everything looks exactly as it did before

## How can we measure success?

We only have to write one set of CSS rules, no matter what direction our language is in.

## Have you considered any issues?

This PR changes the names of all our properties such as `margin-left` but also all our custom variables such as `--quote-padding-left: var(--spacing-larger);` becoming `--quote-padding-inline-start: var(--spacing-larger);`.

### Solutions - 1: Release Notes
We will need to make sure we have release notes for people to update anywhere they have used those variables (it should be a simple enough find/replace task).

### Solutions - 2: Mapped variables
We could create new variables and map those to the old ones, such as:
- `--quote-padding-inline-start: var(--quote-padding-left);`
